flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -146,6 +143,11 @@ public static TypeInfo toHiveTypeInfo(DataType dataType) {
 			// Flink's primitive types that Hive 2.3.4 doesn't support: Time, TIMESTAMP_WITH_LOCAL_TIME_ZONE
 		}
 
+		if (dataType.equals(DataTypes.BYTES())) {
```

Comment:
This way will equals nullable, you should put this code in switch case, and check length.